builder/dockerfile/detectcontext.go

```diff
@@ -0,0 +1,144 @@
+package dockerfile
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/docker/builder"
+	"github.com/docker/docker/builder/dockerignore"
+	"github.com/docker/docker/pkg/fileutils"
+	"github.com/docker/docker/pkg/httputils"
+	"github.com/docker/docker/pkg/symlink"
+	"github.com/docker/docker/pkg/urlutil"
+)
+
+// DetectRemoteContext returns a context and dockerfile from remote location or
+// local archive. progressReader is only used if remoteURL is actually a URL
+// (not empty, and not a Git endpoint).
+func (bm *BuildManager) DetectRemoteContext(ctx context.Context, remoteURL string, dockerfilePath string, r io.ReadCloser, progressReader func(in io.ReadCloser) io.ReadCloser) (remote builder.Remote, dockerfile io.ReadCloser, err error) {
```

Comment:
Looks like `bm` is not used here, so this could be a function instead of an instance method on `buildManager`.